SPEC.md

```diff
@@ -0,0 +1,28 @@
+# Navigation.js specification
+
+## Purpose
+The goal of this specification is to outline the rules and ideas this library adheres to guide a user along a route.
+
+### Units
+By default, the metric system should be used where available.
+
+### Concepts
+Navigation.js looks at a single step along a greater route. All calculations are done against this step and not against the whole route. Once a step is completed, the next step on the route can be compared. If the user is no longer on the current step, the developer should re-route the user and also set their internal step counter back to 0.
+
+### Re-routing
+If at any time the user is not within `45 meters` of any point on the current step, re-route the user.
+
+### Completing a step
+* If the user is within 15 meters of the end of the step, the user has completed the current step.
+* If the bearing for the user is available, the above point must be true and also the users bearing must be within `20 degrees` of the exit bearing of the step.
```

Comment:
What if it's a slight turn or a continue instruction? What if it's a merge to the left onto a road that's curving to the right (a common scenario on highways)?